### PR TITLE
[terra-infinite-list] Update docs to use MDX syntax

### PR DESCRIPTION
**Pull request by saket2403**
### Summary
Updated terra-infinite-list docs to use MDX syntax

### CODE REVIEW

File: packages/terra-infinite-list/src/terra-dev-site/doc/infinite-list/InfiniteList.1.doc.mdx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+## List Item Selection
+
+Management of list selection is provided by the HOC implementing the InfiniteList. For an example of list item selection you can refer to List's documentation:
+
+* [Single Select List](https://engineering.cerner.com/terra-ui/#/components/terra-list/list/guides/single-select-list)
+* [Multi Select List](https://engineering.cerner.com/terra-ui/#/components/terra-list/list/guides/multi-select-list)
+
+[View Component Source Code](https://github.com/cerner/terra-framework/tree/master/packages/terra-hookshot)
+
+
```
Comment (emilyrohrbough) [inline packages/terra-infinite-list/src/terra-dev-site/doc/infinite-list/InfiniteList.1.doc.mdx]: nit: extra line

File: packages/terra-infinite-list/src/terra-dev-site/doc/infinite-list/InfiniteList.1.doc.mdx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+## Examples
+<InitialLoadingExample title='Example Initial Loading Indicator' />
```
Comment (emilyrohrbough) [inline packages/terra-infinite-list/src/terra-dev-site/doc/infinite-list/InfiniteList.1.doc.mdx]: Should these be double quotes?

Comment (saket2403) [reply packages/terra-infinite-list/src/terra-dev-site/doc/infinite-list/InfiniteList.1.doc.mdx]: Single Quotes works fine

File: packages/terra-infinite-list/src/terra-dev-site/doc/infinite-list/InfiniteList.1.doc.mdx (hunk 1 above)
Comment (saket2403) [reply packages/terra-infinite-list/src/terra-dev-site/doc/infinite-list/InfiniteList.1.doc.mdx]: Removed

**Pull request merged**